lib/GameConstants.h

```diff
@@ -957,6 +957,7 @@ typedef si64 TExpType;
 typedef std::pair<ui32, ui32> TDmgRange;
 typedef si32 TBonusSubtype;
 typedef si32 TQuantity;
+typedef std::vector<std::vector<std::vector<ui8>>> TFoWMap;

```

Comment:
If you want to reduce memory footprint for CPU cache then this should be vector of bool's (std::vector<bool> uses 1 bit per value).

Or even some custom class that stores all 3 levels of map in one memory block (there is boost multarray but I don't know how it handles bool's). In this case fow map would be more likely to fit completely into CPU cache ( < 5 kb memory block)